zio-cli/shared/src/main/scala/zio/cli/PrimType.scala

```diff
@@ -36,20 +37,93 @@ object PrimType {
+    def validate(value: String): IO[String, BigDecimal] = attempt(value, BigDecimal(_), "decimal")
+  }
+  
+  case object Integer extends PrimType[BigInt] {
+    def validate(value: String): IO[String, BigInt] = attempt(value, BigInt(_), "integer")
+  }
+
+  case object Boolean extends PrimType[Boolean] {
+    def validate(value: String): IO[String, Boolean] = value match {
+      case "true" => IO.succeed(true)
+      case "false" => IO.succeed(false)
+      case s => IO.fail(s"Couldn't parse Boolean from $s.")
+    }
+  }
+  
+  case object Instant extends PrimType[JInstant] {
+    def validate(value: String): IO[String, JInstant] = attempt(value, JInstant.parse, "instant")
+  }
+
+  case object LocalDate extends PrimType[JLocalDate] {
+    def validate(value: String): IO[String, JLocalDate] = attempt(value, JLocalDate.parse, "localdate")
+  }
+
+  case object LocalDateTime extends PrimType[JLocalDateTime] {
+    def validate(value: String): IO[String, JLocalDateTime] = attempt(value, JLocalDateTime.parse, "localdatetime")
+  }
+  
+  case object LocalTime extends PrimType[JLocalTime] {
+    def validate(value: String): IO[String, JLocalTime] = attempt(value, JLocalTime.parse, "localtime")
+  }
+
+  case object MonthDay extends PrimType[JMonthDay] {
+    def validate(value: String): IO[String, JMonthDay] = attempt(value, JMonthDay.parse, "monthday")
+  }
+
+  case object OffsetDateTime extends PrimType[JOffsetDateTime] {
+    def validate(value: String): IO[String, JOffsetDateTime] = attempt(value, JOffsetDateTime.parse, "offsetdatetime")
+  }
+  
+  case object OffsetTime extends PrimType[JOffsetTime] {
+    def validate(value: String): IO[String, JOffsetTime] = attempt(value, JOffsetTime.parse, "offsettime")
+  }
+  
+  case object Period extends PrimType[JPeriod] {
+    def validate(value: String): IO[String, JPeriod] = attempt(value, JPeriod.parse, "period")
+  }
+  
+  case object Year extends PrimType[JYear] {
+    def validate(value: String): IO[String, JYear] = attempt(value, JYear.parse, "year")
+  }
+  
+  case object YearMonth extends PrimType[JYearMonth] {
+    def validate(value: String): IO[String, JYearMonth] = attempt(value, JYearMonth.parse, "yearmonth")
+  }
+  case object ZonedDateTime extends PrimType[JZonedDateTime] {
+    def validate(value: String): IO[String, JZonedDateTime] = attempt(value, JZonedDateTime.parse, "zoneddatetime")
+  }
+
+  case object ZoneId extends PrimType[JZoneId] {
+    def validate(value: String): IO[String, JZoneId] = attempt(value, JZoneId.of, "zoneid")
+  }
+  case object ZoneOffset extends PrimType[JZoneOffset] {
+    def validate(value: String): IO[String, JZoneOffset] = attempt(value, JZoneOffset.of, "zoneoffset")
+  }
+
+  private def attempt[A, E](value: String, parse: String => A, typeName: String): IO[String, A] =
+    Try(parse(value)) match {
+      case Success(value)     => IO.succeed(value)
+      case Failure(throwable) => IO.fail(s"Couldn't parse $typeName from value: $value.")
```

Comment:
This should ideally be more user friendly, since a user of a CLI will not necessarily be a developer or understand the word "parse".

Reply:
I know this will entail more work, but I think it's worth it

Reply:
Sure, what kind of format do u have in mind here?

Reply:
Something like:

```
"'January' is not a month. A month must have the format 1 - 12.
```